llvm/include/llvm/Transforms/Vectorize/LoopVectorizationLegality.h

```diff
@@ -463,14 +460,10 @@ class LoopVectorizationLegality {
     return CountableExitingBlocks;
   }
 
-  /// Returns all the exiting blocks with an uncountable exit.
-  const SmallVector<BasicBlock *, 4> &getUncountableExitingBlocks() const {
-    return UncountableExitingBlocks;
-  }
-
-  /// Returns all the exit blocks from uncountable exiting blocks.
-  SmallVector<BasicBlock *, 4> getUncountableExitBlocks() const {
-    return UncountableExitBlocks;
+  /// Returns the loop edge with an uncountable exit.
```

Comment:
Would be good to cover also the empty case, perhaps something like

```suggestion
  /// Returns the loop edge with an uncountable exit or std::nullopt if there isn't a single such edge.
```

Reply:
Done